src/python/gudhi/representations/vector_methods.py

```diff
@@ -379,72 +384,80 @@ def transform(self, X):
         if not self.is_fitted():
             raise NotFittedError("Not fitted.")
 
-        if len(X) == 0:
-            X = [np.zeros((0, 2))]
-        
         N = len(X)
 
-        events = np.concatenate([pd.ravel(order="F") for pd in X], axis=0)
-        sorting = np.argsort(events)
-        offsets = np.zeros(1 + N, dtype=int)
-        for i in range(0, N):
-            offsets[i+1] = offsets[i] + 2*X[i].shape[0]
-        starts = offsets[0:N]
-        ends = offsets[1:N + 1] - 1
+        if sum([len(x) for x in X]) == 0:
 
-        bettis = [[0] for i in range(0, N)]
+            return np.zeros((N, self.resolution))
+            
+        else:
+
+            events = np.concatenate([pd.ravel(order="F") for pd in X], axis=0)
+            sorting = np.argsort(events)
+            offsets = np.zeros(1 + N, dtype=int)
+            for i in range(0, N):
+                offsets[i+1] = offsets[i] + 2*X[i].shape[0]
+            starts = offsets[0:N]
+            ends = offsets[1:N + 1] - 1
 
-        i = 0
-        for x in self.grid_:
-            while i < len(sorting) and events[sorting[i]] <= x:
-                j = np.searchsorted(ends, sorting[i])
-                delta = 1 if sorting[i] - starts[j] < len(X[j]) else -1
-                bettis[j][-1] += delta
-                i += 1
-            for k in range(0, N):
-                bettis[k].append(bettis[k][-1])
+            bettis = [[0] for i in range(0, N)]
 
-        return np.array(bettis, dtype=int)[:, 0:-1]
+            i = 0
+            for x in self.grid_:
+                while i < len(sorting) and events[sorting[i]] <= x:
+                    j = np.searchsorted(ends, sorting[i])
+                    delta = 1 if sorting[i] - starts[j] < len(X[j]) else -1
+                    bettis[j][-1] += delta
+                    i += 1
+                for k in range(0, N):
+                    bettis[k].append(bettis[k][-1])
+    
+            return np.array(bettis, dtype=int)[:, 0:-1]
 
     def fit_transform(self, X):
         """
         The result is the same as fit(X) followed by transform(X), but potentially faster.
         """
 
         if self.predefined_grid is None and self.resolution is None:
-            if len(X) == 0:
-                X = [np.zeros((0, 2))]
 
             N = len(X)
 
-            events = np.concatenate([pd.ravel(order="F") for pd in X], axis=0)
-            sorting = np.argsort(events)
-            offsets = np.zeros(1 + N, dtype=int)
-            for i in range(0, N):
-                offsets[i+1] = offsets[i] + 2*X[i].shape[0]
-            starts = offsets[0:N]
-            ends = offsets[1:N + 1] - 1
+            if sum([len(x) for x in X]) == 0:
 
-            xs = [-np.inf]
-            bettis = [[0] for i in range(0, N)]
+                return np.zeros((N, 1))
```

Comment:
Should we still set `self.grid_`? In case we want to call `transform` on this object afterwards. (one of the tests seems to fail because of that)